applets/clock/clock.c

```diff
@@ -2264,11 +2264,12 @@ cities_changed (GSettings    *settings,
 
         cur = mate_panel_applet_settings_get_gslist (settings, key);
 
-        while (cur) {
-                const char *str = cur->data;
+        for (l = cur; l; l = l->next) {
+                char *str = l->data;
                 g_markup_parse_context_parse (context, str, strlen (str), NULL);
-                cur = cur->next;
+                g_free (str);
         }
+        g_slist_free (cur);
```

Comment:
You can clean all after leaving the for lopp:
```
         /* this can be improved by using g_slist_foreach */
        for (l = cur; l; l = l->next) {
                g_markup_parse_context_parse (context, str, strlen (str), NULL);
         }
         g_slist_free_full (cur;)
```

Reply:
Fixed.